File: backend/src/main/java/br/com/asa/services/UserService.java
package br.com.asa.services;

import java.util.List;
import java.util.Optional;

import br.com.asa.dtos.UserDto;
import br.com.asa.models.UserEntity;
import org.springframework.data.domain.Page;

public interface UserService {
	Optional<UserEntity> getUserInformationById(Integer id);
	Page<UserEntity> findAll(Integer pageNo, Integer pageSize, String sortBy);
	List<UserEntity> getByUser(String search);
	boolean emailValidation(String email);
	boolean cpfValidation(String cpf);
	void save(UserDto userDto);
	String getEmailById(int id);
}

File: frontend/src/services/UserService.js
import http from '@/services/http-common'

const path = '/user'

class UserService {
  getAll(currentPage) {
    return http.get(`${path}?pageNo=${currentPage}`)
  }

  getByUser(search) {
    return http.get(`${path}/search?user=${search}`)
  }

  new(data) {
    return http.post(`${path}/new`, data)
  }

  edit(data) {
    return http.put(`${path}/edit`, data)
  }
}

export default new UserService()
